### PR TITLE
Create release.yml

### LINKED ISSUE #68: Set up CI for Jazzband project releases

### PR DESCRIPTION
**Pull request by graingert**
fixes https://github.com/jazzband/pathlib2/issues/68

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          python -m build --sdist --wheel
+          twine check dist/*
+
+      - name: Upload packages to Jazzband
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: jazzband
+          password: ${{ secrets.JAZZBAND_RELEASE_KEY }}
+          repository_url: https://jazzband.co/projects/contextlib2/upload
```
Comment (graingert) [inline .github/workflows/release.yml]:
```suggestion
          repository_url: https://jazzband.co/projects/pathlib2/upload
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+name: Release
+
+on:
+  push:
+    tags:
+    - '*'
+
+jobs:
+  build:
+    if: github.repository == 'jazzband/contextlib2'
```
Comment (graingert) [inline .github/workflows/release.yml]:
```suggestion
    if: github.repository == 'jazzband/pathlib2'
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: release-${{ hashFiles('**/tox.ini') }}
+          restore-keys: |
+            release-
```
Comment (graingert) [inline .github/workflows/release.yml]: I don't think this pip cache is needed - it's only the pip setuptools twine wheel and build deps used.

```suggestion
```

**Comment by mcmtroffaes** [comment]
Looks great, thanks for putting together!

**Pull request merged**